configurablecare/app/src/main/java/com/google/fhir/examples/configurablecare/HomeFragment.kt

```diff
@@ -0,0 +1,109 @@
+ */
+package com.google.fhir.examples.configurablecare
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.MenuItem
+import android.view.View
+import android.view.ViewGroup
+import android.widget.AdapterView
+import android.widget.ArrayAdapter
+import androidx.appcompat.app.AppCompatActivity
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.activityViewModels
+import androidx.navigation.fragment.findNavController
+import com.google.fhir.examples.configurablecare.care.CareConfiguration
+import com.google.fhir.examples.configurablecare.care.CareWorkflowExecutionViewModel
+import com.google.fhir.examples.configurablecare.care.ConfigurationManager
+import com.google.fhir.examples.configurablecare.care.ConfigurationManager.setServiceRequestConfigMap
+import com.google.fhir.examples.configurablecare.care.ConfigurationManager.setTaskConfigMap
+import com.google.fhir.examples.configurablecare.databinding.FragmentHomeBinding
+
+class HomeFragment : Fragment() {
+  private lateinit var careConfiguration: CareConfiguration
+  private val workflowExecutionViewModel by activityViewModels<CareWorkflowExecutionViewModel>()
+  private var _binding: FragmentHomeBinding? = null
+  private val binding
+    get() = _binding!!
+
+  override fun onCreateView(
+    inflater: LayoutInflater,
+    container: ViewGroup?,
+    savedInstanceState: Bundle?
+  ): View {
+    _binding = FragmentHomeBinding.inflate(inflater, container, false)
+    careConfiguration = ConfigurationManager.getCareConfiguration(requireContext())
+    return binding.root
+  }
+
+  override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+    super.onViewCreated(view, savedInstanceState)
+    (requireActivity() as AppCompatActivity).supportActionBar?.apply {
+      title = resources.getString(R.string.app_name)
+      setDisplayHomeAsUpEnabled(true)
+    }
+    setHasOptionsMenu(true)
+    (activity as MainActivity).setDrawerEnabled(true)
+    setIGSpinner()
+    setOnClicks()
+  }
+
+  private fun setIGSpinner() {
+    val igSpinner = binding.igSpinner
+    val adapter =
+      ArrayAdapter<String>(requireContext(), android.R.layout.simple_spinner_dropdown_item)
+    careConfiguration.supportedImplementationGuides.forEach {
+      adapter.add(it.implementationGuideConfig.entryPoint.substring("PlanDefinition/".length))
+    }
+    adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
+    igSpinner.adapter = adapter
+
+    igSpinner.onItemSelectedListener =
+      object : AdapterView.OnItemSelectedListener {
+        override fun onItemSelected(p0: AdapterView<*>?, p1: View?, p2: Int, p3: Long) {
+          if (p0 != null) {
+            workflowExecutionViewModel.currentPlanDefinitionId = p0.getItemAtPosition(p2) as String
+            setTaskConfigMap(workflowExecutionViewModel.currentPlanDefinitionId)
+            setServiceRequestConfigMap(workflowExecutionViewModel.currentPlanDefinitionId)
```

Comment:
`getTaskConfigMap` is called in FhirApplication before `setTaskConfigMap` is called  here. Same is the case with `setServiceRequestConfigMap`. This probably works because of lazy initialisation of the variables. 

But once the TaskManager is lazily initialized in the FhirApplication it won't change the taskConfigMap with which is was initialized even when we change the PlanDefinition and setTaskConfigMap is called, right ?

I think we should re-initialize taskManager every time a new PlanDefinition is selected from the drop down!

Reply:
We will be setting an "active" resourceResourceConfig whenever a PlanDefinition is selected and this will be passed along to the CarePlanManager when $apply needs to be called. The CarePlanManager will pass this config to the TaskManager whenever a Task has to be created.